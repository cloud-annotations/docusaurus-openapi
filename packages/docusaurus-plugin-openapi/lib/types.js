"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = void 0;
var Type;
(function (Type) {
    Type["string"] = "string";
    Type["number"] = "number";
    Type["integer"] = "integer";
    Type["boolean"] = "boolean";
    Type["object"] = "object";
    Type["array"] = "array";
})(Type = exports.Type || (exports.Type = {}));
