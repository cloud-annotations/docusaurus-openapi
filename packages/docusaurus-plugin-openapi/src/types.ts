import { Request } from "postman-collection";

export interface PluginOptions {
  openapiPath: string;
  routeBasePath: string;
  apiLayoutComponent: string;
  apiItemComponent: string;
  remarkPlugins: ([Function, object] | Function)[];
  rehypePlugins: string[];
  admonitions: any;
}

export interface LoadedContent {
  openapiData: any;
}

export enum Type {
  string = "string",
  number = "number",
  integer = "integer",
  boolean = "boolean",
  object = "object",
  array = "array",
}

export interface Schema {
  type?: Type;
  format?: string;
  example?: any;
  additionalProperties?: any;
  enum?: any;
  default?: any;
  deprecated?: boolean;

  properties?: Schema;
  items?: Schema;
  oneOf?: Schema;
  anyOf?: Schema;
}

export interface Primitives {
  string: { [key: string]: (schema: Schema) => any };
  number: { [key: string]: (schema: Schema) => any };
  integer: { [key: string]: (schema: Schema) => any };
  boolean: { [key: string]: (schema: Schema) => any };
  object: { [key: string]: (schema: Schema) => any };
  array: { [key: string]: (schema: Schema) => any };
}

export interface Sidebar {
  [sidebarId: string]: SidebarItem[];
}

export type SidebarItem = SidebarItemLink | SidebarItemCategory;

export interface SidebarItemLink {
  type: "link";
  href: string;
  label: string;
}

export interface SidebarItemCategory {
  type: "category";
  label: string;
  items: SidebarItem[];
  collapsed: boolean;
}

export interface Order {
  [id: string]: OrderMetadata;
}

export interface OrderMetadata {
  previous?: string;
  next?: string;
  sidebar?: string;
}

export interface OpenApiObject {
  openapi: string;
  info: InfoObject;
  servers?: ServerObject[];
  paths: PathsObject;
  components?: ComponentsObject;
  security?: SecurityRequirementObject[];
  tags?: TagObject[];
  externalDocs?: ExternalDocumentationObject;
}

export interface InfoObject {
  title: string;
  version: string;
  description?: string;
  termsOfService?: string;
  contact?: ContactObject;
  license?: LicenseObject;
}

export interface ContactObject {
  name?: string;
  url?: string;
  email?: string;
}

export interface LicenseObject {
  name: string;
  url?: string;
}

export interface ServerObject {
  url: string;
  description?: string;
  variables?: Map<ServerVariable>;
}

export interface ServerVariable {
  default: string;
  enum?: string[];
  description?: string;
}

export interface ComponentsObject {
  schemas?: Map<SchemaObject | ReferenceObject>;
  responses?: Map<ResponseObject | ReferenceObject>;
  parameters?: Map<ParameterObject | ReferenceObject>;
  examples?: Map<ExampleObject | ReferenceObject>;
  requestBodies?: Map<RequestBodyObject | ReferenceObject>;
  headers?: Map<HeaderObject | ReferenceObject>;
  securitySchemes?: Map<
    | ApiKeySecuritySchemeObject
    | HttpSecuritySchemeObject
    | Oauth2SecuritySchemeObject
    | OpenIdConnectSecuritySchemeObject
    | ReferenceObject
  >;
  links?: Map<LinkObject | ReferenceObject>;
  callbacks?: Map<CallbackObject | ReferenceObject>;
}

export interface PathsObject {
  [path: string]: PathItemObject;
}

export interface PathItemObject {
  $ref?: string;
  summary?: string;
  description?: string;
  get?: OperationObject;
  put?: OperationObject;
  post?: OperationObject;
  delete?: OperationObject;
  options?: OperationObject;
  head?: OperationObject;
  patch?: OperationObject;
  trace?: OperationObject;
  servers?: ServerObject[];
  parameters?: (ParameterObject | ReferenceObject)[];
}

export interface OperationObject {
  responses: ResponsesObject;
  tags?: string[];
  summary?: string;
  description?: string;
  externalDocs?: ExternalDocumentationObject;
  operationId?: string;
  parameters?: (ParameterObject | ReferenceObject)[];
  requestBody?: RequestBodyObject | ReferenceObject;
  callbacks?: Map<CallbackObject | ReferenceObject>;
  deprecated?: boolean;
  security?: SecurityRequirementObject[];
  servers?: ServerObject[];
  postman?: Request;
}

export interface ExternalDocumentationObject {
  url: string;
  description?: string;
}

export interface ParameterObject {
  name: string;
  in: string;
  description?: string;
  required?: boolean;
  deprecated?: boolean;
  allowEmptyValue?: boolean;
  //
  style?: string;
  explode?: string;
  allowReserved?: boolean;
  schema?: SchemaObject | ReferenceObject;
  example?: any;
  examples?: Map<ExampleObject | ReferenceObject>;
  //
  content?: Map<MediaTypeObject>;
  // ignoring stylings: matrix, label, form, simple, spaceDelimited,
  // pipeDelimited and deepObject
}

export interface RequestBodyObject {
  content: Map<MediaTypeObject>;
  description?: string;
  required?: boolean;
}

export interface MediaTypeObject {
  schema?: SchemaObject | ReferenceObject;
  example?: any;
  examples?: Map<ExampleObject | ReferenceObject>;
  encoding?: Map<EncodingObject>;
}

export interface EncodingObject {
  contentType?: string;
  headers?: Map<HeaderObject | ReferenceObject>;
  style?: string;
  explode?: boolean;
  allowReserved?: boolean;
}

export interface ResponsesObject {
  [code: string]: ResponseObject | ReferenceObject;
}

export interface ResponseObject {
  description: string;
  headers?: Map<HeaderObject | ReferenceObject>;
  content?: Map<MediaTypeObject>;
  links?: Map<LinkObject | ReferenceObject>;
}

export interface CallbackObject {
  [expression: string]: PathItemObject;
}

export interface ExampleObject {
  summary?: string;
  description?: string;
  value?: any;
  externalValue?: string;
}

export interface LinkObject {
  operationRef?: string;
  operationId?: string;
  parameters?: Map<any>;
  requestBody?: any;
  description?: string;
  server?: ServerObject;
}

export interface HeaderObject {
  description?: string;
  required?: boolean;
  deprecated?: boolean;
  allowEmptyValue?: boolean;
  //
  style?: string;
  explode?: string;
  allowReserved?: boolean;
  schema?: SchemaObject | ReferenceObject;
  example?: any;
  examples?: Map<ExampleObject | ReferenceObject>;
  //
  content?: Map<MediaTypeObject>;
  // ignoring stylings: matrix, label, form, simple, spaceDelimited,
  // pipeDelimited and deepObject
}

export interface TagObject {
  name: string;
  description?: string;
  externalDocs?: ExternalDocumentationObject;
}

export interface ReferenceObject {
  $ref: string;
}

// TODO: this could be expanded on.
export interface SchemaObject {
  [key: string]: any;
}

export interface ApiKeySecuritySchemeObject {
  type: string;
  description?: string;
  name: string;
  in: string;
}

export interface HttpSecuritySchemeObject {
  type: string;
  description?: string;
  scheme: string;
  bearerFormat?: string;
}

export interface Oauth2SecuritySchemeObject {
  type: string;
  description?: string;
  flows: OAuthFlowsObject;
}

export interface OpenIdConnectSecuritySchemeObject {
  type: string;
  description?: string;
  openIdConnectUrl: string;
}

export interface OAuthFlowsObject {
  implicit?: OAuthFlowObject;
  password?: OAuthFlowObject;
  clientCredentials?: OAuthFlowObject;
  authorizationCode?: OAuthFlowObject;
}

export interface OAuthFlowObject {
  authorizationUrl?: string; // required for some?
  tokenUrl?: string; // required for some?
  refreshUrl: string;
  scopes: Map<string>;
}

export interface SecurityRequirementObject {
  [name: string]: string[];
}

export interface Map<T> {
  [key: string]: T;
}

export interface ApiSection {
  title: string;
  description: string;
  items: ApiItem[];
}

export interface ApiItem extends OperationObject {
  method: string;
  path: string;
  hashId: string;
  summary: string;
  permalink: string;
  next: Page;
  previous: Page;
  jsonRequestBodyExample: string;
}

export interface Page {
  title: string;
  permalink: string;
}
