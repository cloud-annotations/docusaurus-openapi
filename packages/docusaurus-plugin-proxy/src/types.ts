/* ============================================================================
 * Copyright (c) Cloud Annotations
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 * ========================================================================== */

import type { Configuration } from "webpack-dev-server";

export interface PluginOptions {
  proxy: Configuration["proxy"];
}
