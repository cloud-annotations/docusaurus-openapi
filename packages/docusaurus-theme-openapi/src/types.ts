/* ============================================================================
 * Copyright (c) Cloud Annotations
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 * ========================================================================== */

export interface ThemeConfig {
  api?: {
    proxy?: string;
    authPersistance?: false | "localStorage" | "sessionStorage";
    serverVariablesPersistance?: false | "localStorage" | "sessionStorage";
  };
}
