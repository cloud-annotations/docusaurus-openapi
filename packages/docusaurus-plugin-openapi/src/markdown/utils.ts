/* ============================================================================
 * Copyright (c) Cloud Annotations
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 * ========================================================================== */

export type Children = string | undefined | (string | undefined)[];

export type Props = Record<string, any> & { children?: Children };

export function create(tag: string, props: Props): string {
  const { children, ...rest } = props;

  let propString = "";
  for (const [key, value] of Object.entries(rest)) {
    propString += ` ${key}={${JSON.stringify(value)}}`;
  }

  return `<${tag}${propString}>${render(children)}</${tag}>`;
}

export function guard<T>(
  value: T | undefined,
  cb: (value: T) => Children
): string {
  if (value) {
    const children = cb(value);
    return render(children);
  }
  return "";
}

export function render(children: Children): string {
  if (Array.isArray(children)) {
    return children.filter((c) => c !== undefined).join("");
  }
  return children ?? "";
}
