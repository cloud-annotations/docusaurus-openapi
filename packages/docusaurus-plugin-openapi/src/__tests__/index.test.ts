/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import {validate} from 'webpack';
import {isMatch} from 'picomatch';
import commander from 'commander';
import fs from 'fs-extra';
import pluginContentDocs from '../index';
import loadEnv from '../env';
import {loadContext} from '@docusaurus/core/src/server/index';
import {applyConfigureWebpack} from '@docusaurus/core/src/webpack/utils';
import {RouteConfig} from '@docusaurus/types';
import {posixPath} from '@docusaurus/utils';
import {sortConfig} from '@docusaurus/core/src/server/plugins';

import * as version from '../version';

const createFakeActions = (
  routeConfigs: RouteConfig[],
  contentDir,
  dataContainer?,
) => {
  return {
    addRoute: (config: RouteConfig) => {
      routeConfigs.push(config);
    },
    createData: async (name, content) => {
      if (dataContainer) {
        dataContainer[name] = content;
      }
      return path.join(contentDir, name);
    },
  };
};

test('site with wrong sidebar file', async () => {
  const siteDir = path.join(__dirname, '__fixtures__', 'simple-site');
  const context = loadContext(siteDir);
  const sidebarPath = path.join(siteDir, 'wrong-sidebars.json');
  const plugin = pluginContentDocs(context, {
    sidebarPath,
  });
  await expect(plugin.loadContent()).rejects.toThrowErrorMatchingSnapshot();
});

describe('empty/no docs website', () => {
  const siteDir = path.join(__dirname, '__fixtures__', 'empty-site');
  const context = loadContext(siteDir);

  test('no files in docs folder', async () => {
    await fs.ensureDir(path.join(siteDir, 'docs'));
    const plugin = pluginContentDocs(context, {});
    const content = await plugin.loadContent();
    const {docsMetadata, docsSidebars} = content;
    expect(docsMetadata).toMatchInlineSnapshot(`Object {}`);
    expect(docsSidebars).toMatchInlineSnapshot(`Object {}`);

    const routeConfigs = [];
    const pluginContentDir = path.join(context.generatedFilesDir, plugin.name);
    const actions = createFakeActions(routeConfigs, pluginContentDir);

    await plugin.contentLoaded({
      content,
      actions,
    });

    expect(routeConfigs).toEqual([]);
  });

  test('docs folder does not exist', async () => {
    const plugin = pluginContentDocs(context, {path: '/path/does/not/exist/'});
    const content = await plugin.loadContent();
    expect(content).toBeNull();
  });
});

describe('simple website', () => {
  const siteDir = path.join(__dirname, '__fixtures__', 'simple-site');
  const context = loadContext(siteDir);
  const sidebarPath = path.join(siteDir, 'sidebars.json');
  const pluginPath = 'docs';
  const plugin = pluginContentDocs(context, {
    path: pluginPath,
    sidebarPath,
    homePageId: 'hello',
  });
  const pluginContentDir = path.join(context.generatedFilesDir, plugin.name);

  test('extendCli - docsVersion', () => {
    const mock = jest.spyOn(version, 'docsVersion').mockImplementation();
    const cli = new commander.Command();
    plugin.extendCli(cli);
    cli.parse(['node', 'test', 'docs:version', '1.0.0']);
    expect(mock).toHaveBeenCalledWith('1.0.0', siteDir, {
      path: pluginPath,
      sidebarPath,
    });
    mock.mockRestore();
  });

  test('getPathToWatch', () => {
    const pathToWatch = plugin.getPathsToWatch();
    const matchPattern = pathToWatch.map((filepath) =>
      posixPath(path.relative(siteDir, filepath)),
    );
    expect(matchPattern).not.toEqual([]);
    expect(matchPattern).toMatchInlineSnapshot(`
      Array [
        "docs/**/*.{md,mdx}",
        "sidebars.json",
      ]
    `);
    expect(isMatch('docs/hello.md', matchPattern)).toEqual(true);
    expect(isMatch('docs/hello.mdx', matchPattern)).toEqual(true);
    expect(isMatch('docs/foo/bar.md', matchPattern)).toEqual(true);
    expect(isMatch('docs/hello.js', matchPattern)).toEqual(false);
    expect(isMatch('docs/super.mdl', matchPattern)).toEqual(false);
    expect(isMatch('docs/mdx', matchPattern)).toEqual(false);
    expect(isMatch('sidebars.json', matchPattern)).toEqual(true);
    expect(isMatch('versioned_docs/hello.md', matchPattern)).toEqual(false);
    expect(isMatch('hello.md', matchPattern)).toEqual(false);
    expect(isMatch('super/docs/hello.md', matchPattern)).toEqual(false);
  });

  test('configureWebpack', async () => {
    const config = applyConfigureWebpack(
      plugin.configureWebpack,
      {
        entry: './src/index.js',
        output: {
          filename: 'main.js',
          path: path.resolve(__dirname, 'dist'),
        },
      },
      false,
    );
    const errors = validate(config);
    expect(errors.length).toBe(0);
  });

  test('content', async () => {
    const content = await plugin.loadContent();
    const {
      docsMetadata,
      docsSidebars,
      versionToSidebars,
      permalinkToSidebar,
    } = content;
    expect(versionToSidebars).toEqual({});
    expect(docsMetadata.hello).toEqual({
      id: 'hello',
      isDocsHomePage: true,
      permalink: '/docs/',
      previous: {
        title: 'baz',
        permalink: '/docs/foo/bazSlug.html',
      },
      sidebar: 'docs',
      source: path.join('@site', pluginPath, 'hello.md'),
      title: 'Hello, World !',
      description: 'Hi, Endilie here :)',
      latestVersionMainDocPermalink: undefined,
    });

    expect(docsMetadata['foo/bar']).toEqual({
      id: 'foo/bar',
      isDocsHomePage: false,
      next: {
        title: 'baz',
        permalink: '/docs/foo/bazSlug.html',
      },
      permalink: '/docs/foo/bar',
      sidebar: 'docs',
      source: path.join('@site', pluginPath, 'foo', 'bar.md'),
      title: 'Bar',
      description: 'This is custom description',
      latestVersionMainDocPermalink: undefined,
    });

    expect(docsSidebars).toMatchSnapshot();

    const routeConfigs = [];
    const dataContainer = {};
    const actions = createFakeActions(
      routeConfigs,
      pluginContentDir,
      dataContainer,
    );

    await plugin.contentLoaded({
      content,
      actions,
    });

    // There is only one nested docs route for simple site
    const baseMetadata = JSON.parse(dataContainer['docs-route-ff2.json']);
    expect(baseMetadata.docsSidebars).toEqual(docsSidebars);
    expect(baseMetadata.permalinkToSidebar).toEqual(permalinkToSidebar);

    // Sort the route config like in src/server/plugins/index.ts for consistent snapshot ordering
    sortConfig(routeConfigs);

    expect(routeConfigs).not.toEqual([]);
    expect(routeConfigs).toMatchSnapshot();
  });
});

describe('versioned website', () => {
  const siteDir = path.join(__dirname, '__fixtures__', 'versioned-site');
  const context = loadContext(siteDir);
  const sidebarPath = path.join(siteDir, 'sidebars.json');
  const routeBasePath = 'docs';
  const plugin = pluginContentDocs(context, {
    routeBasePath,
    sidebarPath,
    homePageId: 'hello',
  });
  const env = loadEnv(siteDir);
  const {docsDir: versionedDir} = env.versioning;
  const pluginContentDir = path.join(context.generatedFilesDir, plugin.name);

  test('extendCli - docsVersion', () => {
    const mock = jest.spyOn(version, 'docsVersion').mockImplementation();
    const cli = new commander.Command();
    plugin.extendCli(cli);
    cli.parse(['node', 'test', 'docs:version', '2.0.0']);
    expect(mock).toHaveBeenCalledWith('2.0.0', siteDir, {
      path: routeBasePath,
      sidebarPath,
    });
    mock.mockRestore();
  });

  test('getPathToWatch', () => {
    const pathToWatch = plugin.getPathsToWatch();
    const matchPattern = pathToWatch.map((filepath) =>
      posixPath(path.relative(siteDir, filepath)),
    );
    expect(matchPattern).not.toEqual([]);
    expect(matchPattern).toMatchInlineSnapshot(`
      Array [
        "docs/**/*.{md,mdx}",
        "versioned_sidebars/version-1.0.1-sidebars.json",
        "versioned_sidebars/version-1.0.0-sidebars.json",
        "versioned_docs/version-1.0.1/**/*.{md,mdx}",
        "versioned_docs/version-1.0.0/**/*.{md,mdx}",
        "sidebars.json",
      ]
    `);
    expect(isMatch('docs/hello.md', matchPattern)).toEqual(true);
    expect(isMatch('docs/hello.mdx', matchPattern)).toEqual(true);
    expect(isMatch('docs/foo/bar.md', matchPattern)).toEqual(true);
    expect(isMatch('sidebars.json', matchPattern)).toEqual(true);
    expect(
      isMatch('versioned_docs/version-1.0.0/hello.md', matchPattern),
    ).toEqual(true);
    expect(
      isMatch('versioned_docs/version-1.0.0/foo/bar.md', matchPattern),
    ).toEqual(true);
    expect(
      isMatch('versioned_sidebars/version-1.0.0-sidebars.json', matchPattern),
    ).toEqual(true);

    // Non existing version
    expect(
      isMatch('versioned_docs/version-2.0.0/foo/bar.md', matchPattern),
    ).toEqual(false);
    expect(
      isMatch('versioned_docs/version-2.0.0/hello.md', matchPattern),
    ).toEqual(false);
    expect(
      isMatch('versioned_sidebars/version-2.0.0-sidebars.json', matchPattern),
    ).toEqual(false);

    expect(isMatch('docs/hello.js', matchPattern)).toEqual(false);
    expect(isMatch('docs/super.mdl', matchPattern)).toEqual(false);
    expect(isMatch('docs/mdx', matchPattern)).toEqual(false);
    expect(isMatch('hello.md', matchPattern)).toEqual(false);
    expect(isMatch('super/docs/hello.md', matchPattern)).toEqual(false);
  });

  test('content', async () => {
    const content = await plugin.loadContent();
    const {
      docsMetadata,
      docsSidebars,
      versionToSidebars,
      permalinkToSidebar,
    } = content;

    // foo/baz.md only exists in version -1.0.0
    expect(docsMetadata['foo/baz']).toBeUndefined();
    expect(docsMetadata['version-1.0.1/foo/baz']).toBeUndefined();
    expect(docsMetadata['foo/bar']).toEqual({
      id: 'foo/bar',
      isDocsHomePage: false,
      permalink: '/docs/next/foo/barSlug',
      source: path.join('@site', routeBasePath, 'foo', 'bar.md'),
      title: 'bar',
      description: 'This is next version of bar.',
      version: 'next',
      sidebar: 'docs',
      next: {
        title: 'hello',
        permalink: '/docs/next/',
      },
    });
    expect(docsMetadata['hello']).toEqual({
      id: 'hello',
      isDocsHomePage: true,
      permalink: '/docs/next/',
      source: path.join('@site', routeBasePath, 'hello.md'),
      title: 'hello',
      description: 'Hello next !',
      version: 'next',
      sidebar: 'docs',
      previous: {
        title: 'bar',
        permalink: '/docs/next/foo/barSlug',
      },
    });
    expect(docsMetadata['version-1.0.1/hello']).toEqual({
      id: 'version-1.0.1/hello',
      isDocsHomePage: true,
      permalink: '/docs/',
      source: path.join(
        '@site',
        path.relative(siteDir, versionedDir),
        'version-1.0.1',
        'hello.md',
      ),
      title: 'hello',
      description: 'Hello 1.0.1 !',
      version: '1.0.1',
      sidebar: 'version-1.0.1/docs',
      previous: {
        title: 'bar',
        permalink: '/docs/foo/bar',
      },
      latestVersionMainDocPermalink: undefined,
    });
    expect(docsMetadata['version-1.0.0/foo/baz']).toEqual({
      id: 'version-1.0.0/foo/baz',
      isDocsHomePage: false,
      permalink: '/docs/1.0.0/foo/baz',
      source: path.join(
        '@site',
        path.relative(siteDir, versionedDir),
        'version-1.0.0',
        'foo',
        'baz.md',
      ),
      title: 'baz',
      description:
        'Baz 1.0.0 ! This will be deleted in next subsequent versions.',
      version: '1.0.0',
      sidebar: 'version-1.0.0/docs',
      next: {
        title: 'hello',
        permalink: '/docs/1.0.0/',
      },
      previous: {
        title: 'bar',
        permalink: '/docs/1.0.0/foo/barSlug',
      },
    });

    expect(docsSidebars).toMatchSnapshot('all sidebars');
    expect(versionToSidebars).toMatchSnapshot(
      'sidebars needed for each version',
    );
    const routeConfigs = [];
    const dataContainer = {};
    const actions = createFakeActions(
      routeConfigs,
      pluginContentDir,
      dataContainer,
    );
    await plugin.contentLoaded({
      content,
      actions,
    });

    // The created base metadata for each nested docs route is smartly chunked/ splitted across version
    const latestVersionBaseMetadata = JSON.parse(
      dataContainer['docs-route-ff2.json'],
    );
    expect(latestVersionBaseMetadata).toMatchSnapshot(
      'base metadata for latest version',
    );
    expect(latestVersionBaseMetadata.docsSidebars).not.toEqual(docsSidebars);
    expect(latestVersionBaseMetadata.permalinkToSidebar).not.toEqual(
      permalinkToSidebar,
    );
    const nextVersionBaseMetadata = JSON.parse(
      dataContainer['docs-next-route-1c8.json'],
    );
    expect(nextVersionBaseMetadata).toMatchSnapshot(
      'base metadata for next version',
    );
    expect(nextVersionBaseMetadata.docsSidebars).not.toEqual(docsSidebars);
    expect(nextVersionBaseMetadata.permalinkToSidebar).not.toEqual(
      permalinkToSidebar,
    );
    const firstVersionBaseMetadata = JSON.parse(
      dataContainer['docs-1-0-0-route-660.json'],
    );
    expect(firstVersionBaseMetadata).toMatchSnapshot(
      'base metadata for first version',
    );
    expect(nextVersionBaseMetadata.docsSidebars).not.toEqual(docsSidebars);
    expect(nextVersionBaseMetadata.permalinkToSidebar).not.toEqual(
      permalinkToSidebar,
    );

    // Sort the route config like in src/server/plugins/index.ts for consistent snapshot ordering
    sortConfig(routeConfigs);

    expect(routeConfigs).not.toEqual([]);
    expect(routeConfigs).toMatchSnapshot();
  });
});
