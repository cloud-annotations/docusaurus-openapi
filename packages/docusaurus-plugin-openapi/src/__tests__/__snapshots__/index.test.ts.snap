// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple website content 1`] = `
Object {
  "docs": Array [
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "collapsed": true,
          "items": Array [
            Object {
              "href": "/docs/foo/bar",
              "label": "Bar",
              "type": "link",
            },
            Object {
              "href": "/docs/foo/bazSlug.html",
              "label": "baz",
              "type": "link",
            },
          ],
          "label": "foo",
          "type": "category",
        },
        Object {
          "href": "https://github.com",
          "label": "Github",
          "type": "link",
        },
        Object {
          "href": "/docs/",
          "label": "Hello, World !",
          "type": "link",
        },
      ],
      "label": "Test",
      "type": "category",
    },
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/",
          "label": "Hello, World !",
          "type": "link",
        },
      ],
      "label": "Guides",
      "type": "category",
    },
  ],
}
`;

exports[`simple website content 2`] = `
Array [
  Object {
    "component": "@theme/DocPage",
    "exact": false,
    "modules": Object {
      "docsMetadata": "~docs/docs-route-ff2.json",
    },
    "path": "/docs",
    "priority": undefined,
    "routes": Array [
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/hello.md",
        },
        "path": "/docs/",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/foo/bar.md",
        },
        "path": "/docs/foo/bar",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/foo/baz.md",
        },
        "path": "/docs/foo/bazSlug.html",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/ipsum.md",
        },
        "path": "/docs/ipsum",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/lorem.md",
        },
        "path": "/docs/lorem",
      },
    ],
  },
]
`;

exports[`site with wrong sidebar file 1`] = `
"Bad sidebars file. The document id 'goku' was used in the sidebar, but no document with this id could be found.
Available document ids=
- foo/bar
- foo/baz
- hello
- ipsum
- lorem"
`;

exports[`versioned website content 1`] = `
Array [
  Object {
    "component": "@theme/DocPage",
    "exact": false,
    "modules": Object {
      "docsMetadata": "~docs/docs-1-0-0-route-660.json",
    },
    "path": "/docs/1.0.0",
    "priority": undefined,
    "routes": Array [
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/versioned_docs/version-1.0.0/hello.md",
        },
        "path": "/docs/1.0.0/",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/versioned_docs/version-1.0.0/foo/bar.md",
        },
        "path": "/docs/1.0.0/foo/barSlug",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/versioned_docs/version-1.0.0/foo/baz.md",
        },
        "path": "/docs/1.0.0/foo/baz",
      },
    ],
  },
  Object {
    "component": "@theme/DocPage",
    "exact": false,
    "modules": Object {
      "docsMetadata": "~docs/docs-next-route-1c8.json",
    },
    "path": "/docs/next",
    "priority": undefined,
    "routes": Array [
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/hello.md",
        },
        "path": "/docs/next/",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/docs/foo/bar.md",
        },
        "path": "/docs/next/foo/barSlug",
      },
    ],
  },
  Object {
    "component": "@theme/DocPage",
    "exact": false,
    "modules": Object {
      "docsMetadata": "~docs/docs-route-ff2.json",
    },
    "path": "/docs",
    "priority": -1,
    "routes": Array [
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/versioned_docs/version-1.0.1/hello.md",
        },
        "path": "/docs/",
      },
      Object {
        "component": "@theme/DocItem",
        "exact": true,
        "modules": Object {
          "content": "@site/versioned_docs/version-1.0.1/foo/bar.md",
        },
        "path": "/docs/foo/bar",
      },
    ],
  },
]
`;

exports[`versioned website content: all sidebars 1`] = `
Object {
  "docs": Array [
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/next/foo/barSlug",
          "label": "bar",
          "type": "link",
        },
      ],
      "label": "Test",
      "type": "category",
    },
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/next/",
          "label": "hello",
          "type": "link",
        },
      ],
      "label": "Guides",
      "type": "category",
    },
  ],
  "version-1.0.0/docs": Array [
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/1.0.0/foo/barSlug",
          "label": "bar",
          "type": "link",
        },
        Object {
          "href": "/docs/1.0.0/foo/baz",
          "label": "baz",
          "type": "link",
        },
      ],
      "label": "Test",
      "type": "category",
    },
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/1.0.0/",
          "label": "hello",
          "type": "link",
        },
      ],
      "label": "Guides",
      "type": "category",
    },
  ],
  "version-1.0.1/docs": Array [
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/foo/bar",
          "label": "bar",
          "type": "link",
        },
      ],
      "label": "Test",
      "type": "category",
    },
    Object {
      "collapsed": true,
      "items": Array [
        Object {
          "href": "/docs/",
          "label": "hello",
          "type": "link",
        },
      ],
      "label": "Guides",
      "type": "category",
    },
  ],
}
`;

exports[`versioned website content: base metadata for first version 1`] = `
Object {
  "docsSidebars": Object {
    "version-1.0.0/docs": Array [
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/1.0.0/foo/barSlug",
            "label": "bar",
            "type": "link",
          },
          Object {
            "href": "/docs/1.0.0/foo/baz",
            "label": "baz",
            "type": "link",
          },
        ],
        "label": "Test",
        "type": "category",
      },
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/1.0.0/",
            "label": "hello",
            "type": "link",
          },
        ],
        "label": "Guides",
        "type": "category",
      },
    ],
  },
  "permalinkToSidebar": Object {
    "/docs/1.0.0/": "version-1.0.0/docs",
    "/docs/1.0.0/foo/barSlug": "version-1.0.0/docs",
    "/docs/1.0.0/foo/baz": "version-1.0.0/docs",
  },
  "version": "1.0.0",
}
`;

exports[`versioned website content: base metadata for latest version 1`] = `
Object {
  "docsSidebars": Object {
    "version-1.0.1/docs": Array [
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/foo/bar",
            "label": "bar",
            "type": "link",
          },
        ],
        "label": "Test",
        "type": "category",
      },
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/",
            "label": "hello",
            "type": "link",
          },
        ],
        "label": "Guides",
        "type": "category",
      },
    ],
  },
  "permalinkToSidebar": Object {
    "/docs/": "version-1.0.1/docs",
    "/docs/foo/bar": "version-1.0.1/docs",
  },
  "version": "1.0.1",
}
`;

exports[`versioned website content: base metadata for next version 1`] = `
Object {
  "docsSidebars": Object {
    "docs": Array [
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/next/foo/barSlug",
            "label": "bar",
            "type": "link",
          },
        ],
        "label": "Test",
        "type": "category",
      },
      Object {
        "collapsed": true,
        "items": Array [
          Object {
            "href": "/docs/next/",
            "label": "hello",
            "type": "link",
          },
        ],
        "label": "Guides",
        "type": "category",
      },
    ],
  },
  "permalinkToSidebar": Object {
    "/docs/next/": "docs",
    "/docs/next/foo/barSlug": "docs",
  },
  "version": "next",
}
`;

exports[`versioned website content: sidebars needed for each version 1`] = `
Object {
  "1.0.0": Set {
    "version-1.0.0/docs",
  },
  "1.0.1": Set {
    "version-1.0.1/docs",
  },
  "next": Set {
    "docs",
  },
}
`;
